File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] ans = {-1, -1};
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i<nums.length; i++){
            if(map.containsKey(target-nums[i])){
                ans[0] = map.get(target-nums[i]);
                ans[1] = i;
            }
            else{
                map.put(nums[i], i);
            }
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        from collections import defaultdict
        hash_map={}
        for i in range(len(nums)):
            hash_map[nums[i]]=i
        print(hash_map)
        for i in range(len(nums)):
            x=target-nums[i]
            if x in hash_map and hash_map[x]!=i :
                return [i,hash_map[x]]